polygon/rest/models/request.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, Optional
+
+X_POLYGON_EDGE_ID = "X-Polygon-Edge-ID"
+X_POLYGON_EDGE_IP_ADDRESS = "X-Polygon-Edge-IP-Address"
+X_POLYGON_EDGE_USER_AGENT = "X-Polygon-Edge-User-Agent"
+
+HEADER = "header"
+
+
+class RequestOptionBuilder:
+    def __init__(
+        self,
+        edge_id: Optional[str] = None,
+        edge_ip_address: Optional[str] = None,
+        edge_user: Optional[str] = None,
+    ):
+        """
+        RequestOptionBuilder is a utility class to build polygon api options used in requests.
+        :param edge_id: is a required Launchpad header. It identifies the Edge User requesting data
+        :param edge_ip_address: is a required Launchpad header. It denotes the originating IP Address of the Edge User
+        :param edge_user: is an optional Launchpad header. It denotes the originating UserAgent of the Edge User requesting data.
+        """
+        self.headers: Optional[Dict[str, str]] = None
+        if edge_id is not None and edge_ip_address is not None:
+            self.edge_headers(
+                edge_id=edge_id, edge_ip_address=edge_ip_address, edge_user=edge_user
+            )
+
+    def edge_headers(
+        self,
+        edge_id: Optional[str] = None,
+        edge_ip_address: Optional[str] = None,
```

Comment:
These two shouldn't be `Optional` or defaulted to `None` right?

Reply:
I made these optional to reuse edge_header function.
it seems to be fine though; removed the optional